### ISSUE TITLE
No native build tools in the Device Agent Docker container

### ISSUE THREAD

**Issue by hardillb** [flowforge/flowforge-device-agent#120]
### Current Behavior

There is no make/python/gcc in the node:16-alpine container

This means that if a snapshot includes any nodes that require native compiled nodes these will fail to install.

### Expected Behavior

Basic build tools to be available:

RUN apk add --no-cache --virtual buildtools build-base linux-headers udev python3

### Steps To Reproduce

NA

### Environment

- FlowForge version: 1.8
- Node.js version: 16
- npm version: 8
- Platform/OS: docker/linux
- Browser: NA

**Comment by Steve-Mcl** [comment]
Ben, would that also bring `git` to the container?

**Comment by hardillb** [comment]
no

**Comment by Steve-Mcl** [comment]
Nice short answer.

Let me try again 🤣 

Would we not want `git` included (for instances where a module installs directly from git) or for other (user) applications that might want git in the image?

**Comment by Steve-Mcl** [comment]
And by user applications I mean node-based, node-red based, exec-based applications.

**Comment by hardillb** [comment]
> Would we not want git included (for instances where a module installs directly from git) or for other (user) applications that might want git in the image?

Can you actually specify a git install at the moment?

> And by user applications I mean node-based, node-red based, exec-based applications.

Those would most likely require a custom container already, so the user could add git at that point.

**Comment by Steve-Mcl** [comment]
> > Would we not want git included (for instances where a module installs directly from git) or for other (user) applications that might want git in the image?
> 
> Can you actually specify a git install at the moment?

Some modules specify dependencies from git.  [Docs](https://docs.npmjs.com/cli/v8/configuring-npm/package-json#git-urls-as-dependencies)

Also, it is not beyond the realms of feasibility that a company internal application would require git or a custom module that depends on a git url.

Also, this simplifies #107

**Comment by Steve-Mcl** [comment]
Also, if including build tools, why would you actually _not_ include git?
